src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs

```diff
@@ -394,37 +377,37 @@ private void DoCheckTimeout()
         /// then to leave runtextpos at the ending position. It should leave
         /// runtextpos where it started if there was no match.
         /// </summary>
-        protected abstract void Go();
+        protected virtual void Go() => throw new NotImplementedException();
 
         /// <summary>
         /// The responsibility of FindFirstChar() is to advance runtextpos
         /// until it is at the next position which is a candidate for the
         /// beginning of a successful match.
         /// </summary>
-        protected abstract bool FindFirstChar();
+        protected virtual bool FindFirstChar() => throw new NotImplementedException();
 
         /// <summary>
         /// InitTrackCount must initialize the runtrackcount field; this is
         /// used to know how large the initial runtrack and runstack arrays
         /// must be.
         /// </summary>
-        protected abstract void InitTrackCount();
+        protected virtual void InitTrackCount() => runtrackcount = 0;
 
         /// <summary>
         /// Initializes all the data members that are used by Go()
         /// </summary>
-        private void InitializeForGo()
+        internal void InitializeForGo()
         {
             if (runmatch is null)
             {
                 // Use a hashtabled Match object if the capture numbers are sparse
                 runmatch = runregex!.caps is null ?
-                    new Match(runregex, runregex.capsize, runtext!, runtextbeg, runtextend - runtextbeg, runtextstart) :
-                    new MatchSparse(runregex, runregex.caps, runregex.capsize, runtext!, runtextbeg, runtextend - runtextbeg, runtextstart);
+                    new Match(runregex, runregex.capsize, runtext ?? string.Empty, runtextbeg, runtextend - runtextbeg, runtextstart) :
```

Comment:
Earlier you made the runtext string argument be nullable... why does this then prevent null from being passed in?

Reply:
I need to check again, but I believe I added this because a unit test was failing which was checking the equivalence of Match.Empty and a failed match, or something like that.